### ISSUE TITLE
invalidcastexception initializing intput system

### ISSUE THREAD

**Issue by david-c-kline** [microsoft/MixedRealityToolkit-Unity#2979]
**Does this affect the legacy HoloToolkit (master) or the Mixed Reality Toolkit (mrtk_release)?**
Mixed Reality Toolkit

**Describe the bug**
```
Exception thrown: 'System.InvalidCastException' in Microsoft.MixedReality.Toolkit.dll
'MixedRealityToolkit.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\Users\DefaultAccount\AppData\Local\DevelopmentFiles\Microsoft.MixedReality.ToolkitVS.Release_x86.davidkl\System.Diagnostics.StackTrace.dll'. Cannot find or open the PDB file.
InvalidCastException: Unable to cast object of type 'Microsoft.MixedReality.Toolkit.SDK.Input.MixedRealityInputManager' to type 'Microsoft.MixedReality.Toolkit.Core.Interfaces.IMixedRealityExtensionService'.
   at Microsoft.MixedReality.Toolkit.Core.Services.MixedRealityToolkit.RegisterService(Type type, IMixedRealityService service)
   at Microsoft.MixedReality.Toolkit.Core.Services.MixedRealityToolkit.Initialize()
   at Microsoft.MixedReality.Toolkit.Core.Services.MixedRealityToolkit.InitializeInternal()
   at Microsoft.MixedReality.Toolkit.Core.Services.MixedRealityToolkit.Awake()
   at Microsoft.MixedReality.Toolkit.Core.Services.MixedRealityToolkit.$Invoke2(Int64 instance, Int64* args)
   at UnityEngine.Internal.$MethodUtility.InvokeMethod(Int64 instance, Int64* args, IntPtr method) 
(Filename: <Unknown> Line: 0)
```

**To Reproduce**
1. Checkout mrtk_development branch
2. Load one of the demo scenes (I used the standard materials scene)
3. Build and run (I deployed to a HoloLens using the .net backend)

**Expected behavior**
The system to be correctly initialized

**Actual behavior**
The above exception

**Unity Editor Version**
2018.2.13f1

**Mixed Reality Toolkit Release Version**
mrtk _development branch as of 10/25/2018 11am

**Comment by davidkline-ms** [comment]
It appears that we are landing in the else case of IsCoreSystem and attempting to cast to an IMixedRealityExtensionService on line 533

                MixedRealityComponents.Add(new Tuple<Type, IMixedRealityExtensionService>(type, (IMixedRealityExtensionService)service));

**Comment by StephenHodgson** [comment]
What about il2cpp builds?

.NET isn't officially supported.

**Comment by SimonDarksideJ** [comment]
I'm concerned by this line

```
Unable to cast object of type 'Microsoft.MixedReality.Toolkit.SDK.Input.MixedRealityInputManager' to type 'Microsoft.MixedReality.Toolkit.Core.Interfaces.IMixedRealityExtensionService'
```

Even more concerned as the same error isn't showing up in the Editor

**Comment by StephenHodgson** [comment]
Yeah I'm not sure how `IsCoreSystem` could be failing. Strange.

**Comment by davidkline-ms** [comment]
> What about il2cpp builds?

Same behavior, system loading is getting aborted, HoloLens and UWP immersive,

> .NET isn't officially supported.

It is supported as long as Unity supports it (2 years of 2018 LTS)

**Comment by davidkline-ms** [comment]
> Can you also replicate this on Windows 10 .NET or IL2CCP builds @davidkline-ms ?

Yes, I can repro it on Windows 10 UWP (IL2CPP)

**Comment by StephenHodgson** [comment]
Does it break for PC OpenVR builds as well?

**Comment by davidkline-ms** [comment]
> Does it break for PC OpenVR builds as well?

yes

**Comment by StephenHodgson** [comment]
I wasn't able to reproduce with the steps provided. Is there anything else we might be missing to reproduce this?

**Comment by davidkline-ms** [comment]
Those are the steps that repro'd for me. I have tried multiple (5+) clones and have hit this in each one. If I roll back prior to the rename, the errors do not occur. I am using the same version of Visual Studio and Unity. The only difference is whether or not I have the change.

Tested:
* IL2CPP: HoloLens, WMR Immersive, OpenVR
* .net: HoloLens, WMR Immersive, OpenVR,

I am going to continue investigating. 

This **must** be fully understood before beta is declared.
